src/App.tsx

```diff
@@ -3,29 +3,115 @@ import './App.scss';
 import './styles/general.scss';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
+import { getTodos } from './api';
 
 interface State {
   selectedUserId: number;
+  todos: Todo[],
+  titleQuery: string,
+  statusQuery: string,
 }
 
 class App extends React.Component<{}, State> {
   state: State = {
     selectedUserId: 0,
+    todos: [],
+    titleQuery: '',
+    statusQuery: '',
+  };
+
+  async componentDidMount() {
+    const todosFromServer = await getTodos();
+
+    this.setState({
+      todos: todosFromServer,
+    });
+  }
+
+  handleStatus = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({ statusQuery: event.currentTarget.value });
+  };
+
+  handleQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ titleQuery: event.target.value });
+  };
+
+  changeTodoStatus = (id: number) => {
+    const todoCopy = this.state.todos.map(todo => {
+      if (todo.id === id) {
+        return { ...todo, completed: !todo.completed };
+      }
+
+      return todo;
+    });
+
+    this.setState(() => ({
+      todos: todoCopy,
+    }));
+  };
+
+  selectUser = (userId: number) => {
+    this.setState({
+      selectedUserId: userId,
+    });
+  };
+
+  clearUserSelection = () => {
+    this.setState({
+      selectedUserId: 0,
+    });
+  };
+
+  getPreparedTodos = () => {
+    const { todos, titleQuery, statusQuery } = this.state;
+    let isCompletedStatus: boolean;
+
+    switch (statusQuery) {
+      case 'active':
+        isCompletedStatus = false;
+        break;
+      case 'completed':
+        isCompletedStatus = true;
+        break;
+      default:
+        return todos.filter(todo => (
+          todo.title.toLowerCase()
+            .includes(titleQuery.toLowerCase())
```

Comment:
create a separate variable for it `titleQuery.toLowerCase()`, no need to do it on every iteration